Bump exec-maven-plugin to 3.1.0 in fluss-dist.

### Search before asking

- [x] I searched in the [issues](https://github.com/apache/fluss/issues) and found nothing similar.


### Description

Currently, version of exec-maven-plugin in root  pom.xml  is 3.1.0, while version in fluss-dist is still 1.5.0.

```shell
[INFO] --- exec:1.5.0:exec (create-build-target-link) @ fluss-dist ---
[ERROR] Command execution failed.
java.io.IOException: Stream closed
at java.lang.ProcessBuilder$NullOutputStream.write (ProcessBuilder.java:442)
at java.io.OutputStream.write (OutputStream.java:157)
at java.io.BufferedOutputStream.flushBuffer (BufferedOutputStream.java:81)
at java.io.BufferedOutputStream.flush (BufferedOutputStream.java:142)
at java.io.FilterOutputStream.close (FilterOutputStream.java:182)

```

### Willingness to contribute

- [x] I'm willing to submit a PR!

--- a/fluss-dist/pom.xml
+++ b/fluss-dist/pom.xml
@@ -154,24 +154,23 @@
             <build>
                 <plugins>
                     <plugin>
+                        <artifactId>maven-clean-plugin</artifactId>
+                        <configuration>
+                            <filesets>
+                                <fileset>
+                                    <directory>${project.basedir}/..</directory>
+                                    <includes>
+                                        <include>build-target</include>
+                                    </includes>
+                                </fileset>
+                            </filesets>
+                        </configuration>
+                    </plugin>
+                    <plugin>
                         <groupId>org.codehaus.mojo</groupId>
                         <artifactId>exec-maven-plugin</artifactId>
-                        <version>1.5.0</version>
+                        <version>3.1.0</version>
                         <executions>
-                            <execution>
-                                <id>remove-build-target-link</id>
-                                <phase>clean</phase>
-                                <goals>
-                                    <goal>exec</goal>
-                                </goals>
-                                <configuration>
-                                    <executable>rm</executable>
-                                    <arguments>
-                                        <argument>-f</argument>
-                                        <argument>${project.basedir}/../build-target</argument>
-                                    </arguments>
-                                </configuration>
-                            </execution>
                             <execution>
                                 <id>create-build-target-link</id>
                                 <phase>package</phase>

